### PR TITLE
Fixing imagesdir support; adding prettify support; updating libraries.

### LINKED ISSUE #14: Images support for GitHub File
I try to render the `example-manual.adoc` page from the maven example repository with `gist.asciidoctor.org`.

Url of the file:

    https://github.com/asciidoctor/asciidoctor-maven-examples/blob/master/asciidoc-to-html-example/src/docs/asciidoc/example-manual.adoc

Links:

* [page on GitHub](https://github.com/asciidoctor/asciidoctor-maven-examples/blob/master/asciidoc-to-html-example/src/docs/asciidoc/example-manual.adoc)
* [page rendered with gist.asciidoctor.org](http://gist.asciidoctor.org/?github-asciidoctor%2Fasciidoctor-maven-examples%2F%2Fasciidoc-to-html-example%2Fsrc%2Fdocs%2Fasciidoc%2Fexample-manual.adoc)

In the `Images` section, the image is broken. It looks like this:

Images section screenshot

The `src` attribute of the `img` tag is wrong:

    <h2 id="_images"><a class="anchor" href="#_images"></a>Images</h2>
    <div class="sectionbody">
    <div class="imageblock thumb">
    <div class="content">
    
    </div>
    </div>
    </div>
    </div>

It should be:

### PR DESCRIPTION
**Pull request by nawroth**
Fixes mimetyoe of SVG images by using rawgit.com.
Fixes #14 (see the discussion there)

### CODE REVIEW

File: js/docgist.js [hunk 1]
```diff
@@ -20,18 +16,39 @@ function DocGist($) {
         $gistId.keydown(gist.readSourceId);
     });
 
-    function renderContent(content, link) {
+    function getAsciidoctorOptions() {
+        return Opal.hash2(['to_file', 'safe', 'attributes'], {
```
Comment (mojavelinux) [inline js/docgist.js]: We've learned recently that it is possible to use Opal.hash and then pass in a regular json notation.

```
return Opal.hash({to_file: false, safe: 'secure', attributes: ...})
```

We can't quite figure out why we thought we had to use hash2, other than it was a programming error that got propagated. Both ways work, but clearly Opal.hash is just simpler :)

File: js/docgist.js [hunk 2]
```diff
@@ -20,18 +16,39 @@ function DocGist($) {
...
         var doc, html = undefined;
-        var highlightSource = false;
+        var highlighter = undefined;
         try {
-            doc = Opal.Asciidoctor.$load(content, ASCIIDOCTOR_OPTIONS);
-            var attributeEq = doc.$method('attr?');
-            if (attributeEq.$call('source-highlighter', 'highlightjs') || attributeEq.$call('source-highlighter', 'highlight.js')) {
-              highlightSource = true;
+            doc = Opal.Asciidoctor.$load(content, getAsciidoctorOptions('parse_header_only=true'));
+            console.log(doc);
```
Comment (mojavelinux) [inline js/docgist.js]: Perhaps only log the document when a special debug mode is set, perhaps via the URL?

**Comment by mojavelinux** [comment]
Nice job!

**Comment by mojavelinux** [comment]
> Fixes mimetyoe of SVG images by using rawgit.com.

I wonder if this problem will be solved by the following change: https://github.com/asciidoctor/asciidoctor/issues/1301

Btw, it seems that caching isn't an issue if you use rawgit.com, but would be an issue if you use cdn.rawgit.com.

> “New changes you push to GitHub will be reflected within minutes.“ (when using rawgit.com)

File: js/docgist.js (hunk 2 above)
Comment (nawroth) [inline js/docgist.js]: Oops, forgot that one!

**Comment by nawroth** [comment]
It seems like the mimetype sent by the server takes precedence, so that change wouldn't make any difference.

Regarding rawgit, they don't want production systems to use the non-cdn URLs. So in case a page gets really many hits, that might be a problem. Maybe we should take care of that when we actually have that problem though.

**Comment by mojavelinux** [comment]
> It seems like the mimetype sent by the server takes precedence, so that change wouldn't make any difference.

Thanks for the clarification.

> Maybe we should take care of that when we actually have that problem though.

Agreed. They are probably talking about Hacker News front-page order of hits...so unless people are craving a page of documentation that bad, I think we'll probably be okay with rawgit.com :)

**Comment by nawroth** [comment]
We had a graphgist on the HN front page, generating tons of hits!

**Comment by mojavelinux** [comment]
> We had a graphgist on the HN front page, generating tons of hits!

:open_mouth:

**Pull request merged**